gdnative-core/src/core_types/geom/basis.rs

```diff
@@ -247,9 +247,37 @@ impl Basis {
         copy
     }
 
+    /// Returns linear interpolation on a sphere between two basis by weight amount (on the range of 0.0 to 1.0).
+    #[inline]
+    pub fn slerp(&self, other:Basis, weight:f32) -> Self {
+        #[inline]
+        fn lerp_f(from:f32, to:f32, weight:f32) -> f32 {
+            from + (to - from) * weight
+        }
+        let from = self.to_quat();
+        let to = other.to_quat();
+        let mut result = Basis::from_quat(from.slerp(to, weight));
+        result.elements[0] *= lerp_f(self.elements[0].length(), other.elements[0].length(), weight);
+        result.elements[1] *= lerp_f(self.elements[1].length(), other.elements[1].length(), weight);
+        result.elements[2] *= lerp_f(self.elements[2].length(), other.elements[2].length(), weight);
+
+        result
+    }
+
+    /// Returns linear interpolation between two basis by weight amount (on the range of 0.0 to 1.0).
+    #[inline]
+    pub fn linear_interpolate(&self, other:Basis, weight:f32) -> Self {
```

Comment:
GDScript [doesn't provide this method](https://docs.godotengine.org/en/stable/classes/class_basis.html), but C++ does. Is it useful for a basis?

If yes, I think we should name it `lerp` despite the discrepancy with `Vector3::linear_interpolate` (which was renamed to `lerp` in Godot 4 -- we could also do that, outside this PR).